CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+# How to contribute
+
+This module was developed in to support Puppet training. It is open to
+the public so that students can see the code that goes in to creating
+the classroom experience. Although most of the development is done by 
+Puppet employees we welcome outside contributions. Be aware that your
+suggested change may not fit in with our future plans, so even very good
+code may not be accepted.
+
+Priority will be given to contributions from Puppet Professional Services
+engineers and Service Delivery Partners actively teaching classes.
+
+There are a few guidelines that we need contributors to follow so that 
+we can have a chance of keeping on top of things.
+
+## Getting Started
+
+* Make sure you have a [Jira account](https://tickets.puppetlabs.com)
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue, assuming one does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub
+
+## Making Changes
+
+Releases are made from the `release` branch, which should always be in a
+releasable state. `master` is the active development branch and changes 
+are periodically moved to the release branch for acceptance testing.
+
+Before you begin working on an issue or feature, check that it hasn't
+already been addressed in the `master` branch.
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * Only target release branches if you are certain your fix must be on that
+    branch.
+  * To quickly create a topic branch based on master; `git checkout -b
+    fix/master/my_contribution master`. Please avoid working directly on the
+    `master` branch.
+* Make commits of logical units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the proper format.
```

Comment:
* Don't update version numbers or changelogs. Instead, write a good commit subject and description. We will use the commit subject to update the changelog as needed.

Do you like that @samuelson ?